### PR TITLE
Fix/issue 164

### LINKED ISSUE #166: Resolve issues with PR #164
There are some outstanding issues in #164 to be resolved

### PR DESCRIPTION
**Pull request by tdziurko**
Fixes remaining issues from Pull Request #164, simplifies throttling in `MailgunSender`, closes #166

### CODE REVIEW

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/MailgunSender.java [hunk 1]
```diff
@@ -33,13 +36,17 @@ public MailgunSender() {
...
+                         @Value("${mailing.fromEmail}") String senderAddress,
+                         @Value("${mailing.throttleDelayInSeconds}") long throttleTimeInSeconds) {
         this.mailingRestClient = mailingRestClient;
         this.senderAddress = senderAddress;
+        this.rateLimiter = RateLimiter.create(1.0 / throttleTimeInSeconds);
     }
 
     @Override
     public void sendEmail(String recipientAddress, String subject, String htmlContent) {
+        double timeToAcquire = rateLimiter.acquire();
```
Comment (goostleek) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/MailgunSender.java]: TIL about [RateLimiter](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/util/concurrent/RateLimiter.html) :)

**Comment by tdziurko** [comment]
@delor, @cslysy do you want to add something or comment this PR?

**Comment by cslysy** [comment]
@tdziurko Nope, looks fine for me :+1:

**Comment by delor** [comment]
Better. The only problem I see might be that you are blocking thread that invoked sending emails.

**Pull request merged**